### DISCUSSION TITLE
Get Front-Matter Content

### DISCUSSION THREAD

**Discussion by entity** [nuxt/content#1066] [Q&A] [answered]
Hi all,

I've started tinkering with the new v2 version for Nuxt 3, and it's working fantastic, but I'm wondering how I can fetch additional blog content (from my markdown files).

```js
const { data: navigation } = await useAsyncData('navigation', () => {
  return fetchContentNavigation()
})
```

This returns the `title` and `slug`, but what about additional content that is entered into my markdown file? If this doesn't exist, is it possible somehow we can specify our "schema", as I want to specify author, images etc.

Thanks

**Comment by nozomuikuta** [comment, marked as answer]
@heychazza 

I've not yet tried Nuxt Content v2 deeply, but here are some information for you based on my source code reading.
I hope this helps you somehow.

First, Nuxt Content v2 seems to have `content.navigation.fields` option that let you define what fields of documents to fetch when it creates navigation items to fetch.

https://github.com/nuxt/content/blob/bc67b7708086f20b2d2ca5cd3a328d260909c1d3/src/module.ts#L138-L145

https://github.com/nuxt/content/blob/bc67b7708086f20b2d2ca5cd3a328d260909c1d3/src/runtime/server/navigation.ts#L9-L29

Second, it also seems that you can pass `queryBuilder` to `fetchContentNavigation` function as when you fetch contents (if I'm correct, fetching content navigation is actually a pattern of fetching with predefined queries and some dedicated logic).

https://github.com/nuxt/content/blob/bc67b7708086f20b2d2ca5cd3a328d260909c1d3/src/runtime/composables/navigation.ts#L6-L24

Third, although I'm not sure, you might be able to extend interface `NavItem` as you like, because otherwise `fetchContentNavigation` always returns an array of objects whose type is hard-corded and unchangeable.
@Tahul will know better about this point.

https://github.com/nuxt/content/blob/bc67b7708086f20b2d2ca5cd3a328d260909c1d3/src/runtime/types.d.ts#L193-L201

**Reply by entity** [reply]
> @heychazza
> 
> I've not yet tried Nuxt Content v2 deeply, but here are some information for you based on my source code reading. I hope this helps you somehow.
> 
> First, Nuxt Content v2 seems to have `content.navigation.fields` option that let you define what fields of documents to fetch when it creates navigation items to fetch.
> 
> https://github.com/nuxt/content/blob/bc67b7708086f20b2d2ca5cd3a328d260909c1d3/src/module.ts#L138-L145
> 
> https://github.com/nuxt/content/blob/bc67b7708086f20b2d2ca5cd3a328d260909c1d3/src/runtime/server/navigation.ts#L9-L29
> 
> Second, it also seems that you can pass `queryBuilder` to `fetchContentNavigation` function as when you fetch contents (if I'm correct, fetching content navigation is actually a pattern of fetching with predefined queries and some dedicated logic).
> 
> https://github.com/nuxt/content/blob/bc67b7708086f20b2d2ca5cd3a328d260909c1d3/src/runtime/composables/navigation.ts#L6-L24
> 
> Third, although I'm not sure, you might be able to extend interface `NavItem` as you like, because otherwise `fetchContentNavigation` always returns an array of objects whose type is hard-corded and unchangeable. @Tahul will know better about this point.
> 
> https://github.com/nuxt/content/blob/bc67b7708086f20b2d2ca5cd3a328d260909c1d3/src/runtime/types.d.ts#L193-L201

Awesome, this worked! Thank you.

Only last issue I've found, when clicking to go to a blog I get a white page until I refresh, wasn't sure if this was a dev thing.

**Comment by nozomuikuta** [comment]
Oh, I should have been more precise.

The `NavItem` is an _interface_ and not a _type_, so `$fetch<Array<NavItem>>(...)` itself is totally fine in terms of type-safety.
What I talked about is "when we want to specify the part of `[key: string]: any` of the interface."
